### PR TITLE
CanGc changes from fontfaceset.rs

### PR DESCRIPTION
**Pull request by chickenleaf**
**Note:**
- left a few `CanGc::note()` to be picked up later on.
- in `cssstyledeclaration.rs` line 268 left a `CanGc::note()` since it needed too many changes. Will pick up from there soon.
- `ScrollTo` and `ScrollTo_` entry in `Bindings.conf` under `Window` and `Element` not quite showing up in the generated code, could be a me problem. Will pick these up again.
-----------------

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of #33683

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they do not modify functionality

### CODE REVIEW

**Review by jdm — APPROVED** [review]
I had no idea the fontfaceset changes would be so infectious! Fascinating!

File: components/script/dom/bindings/codegen/Bindings.conf [hunk 1]
```diff
@@ -188,6 +192,13 @@ DOMInterfaces = {
     'canGc': ['Content'],
 },
 
+'HTMLElement': {
+    'canGc': ['GetOffsetParent', 'OffsetTop', 'OffsetLeft', 'OffsetWidth', 'OffsetHeight', 'InnerText', 'GetOuterText', 'Focus', 'Blur'],
+},
+
+'HTMLImageElement': {
+    'canGc': ['Width', 'Height'],
```
Comment (jdm) [inline components/script/dom/bindings/codegen/Bindings.conf:200]:
```suggestion
    'canGc': ['Width', 'Height'],
}
```

**Review by jdm — APPROVED** [review]

**Comment by chickenleaf** [comment]
Why won't this merge 😭😭

**Comment by mukilan** [comment]
> Why won't this merge 😭😭

There are merge conflicts caused by another PR that got merged while this one was queued. Those need to be resolved and pushed to this branch before we can try merging again.

**Comment by chickenleaf** [comment]
> > Why won't this merge 😭😭
> 
> There are merge conflicts caused by another PR that got merged while this one was queued. Those need to be resolved and pushed to this branch before we can try merging again.

this is the third or forth time around this is happening for this, i shouldn't be concerned right? 😭

**Comment by jdm** [comment]
No, it's just that other cangc prs keep merging that touch the same areas because this one is so big.

**Comment by chickenleaf** [comment]
> No, it's just that other cangc prs keep merging that touch the same areas because this one is so big.

ill make sure to send in smaller changes next, i think this will be an issue with `node.rs` as well. will send in when i hit the 200 lines changed mark. In this case and `node.rs` i kept going because i noticed over 1000+ changes getting merged without further commits in some other PRs.

**Pull request merged**